block/header_cache.go

```diff
@@ -0,0 +1,65 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/rollkit/rollkit/types"
+)
+
+// HeaderCache maintains headers that are seen and hard confirmed
+type HeaderCache struct {
+	headers    *sync.Map
+	hashes     *sync.Map
+	daIncluded *sync.Map
+}
+
+// NewHeaderCache returns a new HeaderCache struct
+func NewHeaderCache() *HeaderCache {
+	return &HeaderCache{
+		headers:    new(sync.Map),
+		hashes:     new(sync.Map),
+		daIncluded: new(sync.Map),
+	}
+}
+
+func (hc *HeaderCache) getHeader(height uint64) (*types.SignedHeader, bool) {
+	header, ok := hc.headers.Load(height)
+	if !ok {
+		return nil, false
+	}
+	return header.(*types.SignedHeader), true
+}
+
+func (hc *HeaderCache) setHeader(height uint64, header *types.SignedHeader) {
+	if header != nil {
+		hc.headers.Store(height, header)
+	}
+}
```

Comment:
Slightly different than the data comment. 

The header can't be nil right? So trying to store a nil header is an error. 
That feels likes something we should handle.

Reply:
we can allow setting anything and check for it in the callsite?